ODIN_II/SRC/subtractions.cpp

```diff
@@ -555,15 +575,36 @@ void split_adder_for_sub(nnode_t* nodeo, int a, int b, int sizea, int sizeb, int
     //connect_nodes(node[count - 1], (node[(count - 1)]->num_output_pins - 1), netlist->gnd_node, 0);
     //}
 
-    /* Probably more to do here in freeing the old node! */
-    vtr::free(nodeo->name);
-    vtr::free(nodeo->input_port_sizes);
-    vtr::free(nodeo->output_port_sizes);
+    /* Freeing the old node! */
```

Comment:
This routine is about 250 lines long ... would be better to shorten where possible.  This code addition looks like it could be a helper function that you call with only nodeo passed in.  That would make it a great candidate for a helper function. Any thoughts?

Reply:
We already have a `free_nnode(node)` helper function in netlist utils; this situation is pretty specific for the freeing process since there is weird usage of pointers. However, I couldn't agree more with what you said. I will create a short helper function for this specific file.